bioimageio_core_version: 0.5.3.post2
bioimageio_spec_version: 0.4.5.post4
rdf_sha256: 4cbd435f08e6a8b846105b33d642d29028d01ad2feec7b3e511333cc8b2e406d
status: passed
tests:
  bioimageio:
  - bioimageio_spec_version: 0.4.5post4
    error: null
    name: bioimageio.spec static validation of model RDF 0.4.5 with update to latest
      format version
    nested_errors: {}
    status: passed
    traceback: null
    warnings:
      tags: 'Missing tags for categories: [{''modality'': [''electron-microscopy'',
        ''cryo-electron-microscopy'', ''fluorescence-light-microscopy'', ''transmission-light-microscopy'',
        ''super-resolution-microscopy'', ''x-ray-microscopy'', ''force-microscopy'',
        ''high-content-imaging'', ''whole-slide-imaging'', ''other'']}, {''dims'':
        [''2d'', ''3d'', ''2d-t'', ''3d-t'']}, {''framework'': [''tensorflow'', ''pytorch'',
        ''tensorflow.js'']}, {''software'': [''ilastik'', ''imagej'', ''fiji'', ''imjoy'',
        ''deepimagej'', ''n2v'']}, {''network'': [''unet'', ''densenet'', ''resnet'',
        ''inception'', ''shufflenet'']}, {''task'': [''semantic-segmentation'', ''instance-segmentation'',
        ''object-detection'', ''image-classification'', ''denoising'', ''image-restoration'',
        ''image-reconstruction'', ''in-silico-labeling'']}]'
      weights: {onnx: missing 'opset_version'}
  - bioimageio_spec_version: 0.4.5post4
    error: null
    name: bioimageio.spec static validation of model RDF 0.4.5
    nested_errors: {}
    status: passed
    traceback: null
    warnings:
      tags: 'Missing tags for categories: [{''modality'': [''electron-microscopy'',
        ''cryo-electron-microscopy'', ''fluorescence-light-microscopy'', ''transmission-light-microscopy'',
        ''super-resolution-microscopy'', ''x-ray-microscopy'', ''force-microscopy'',
        ''high-content-imaging'', ''whole-slide-imaging'', ''other'']}, {''dims'':
        [''2d'', ''3d'', ''2d-t'', ''3d-t'']}, {''framework'': [''tensorflow'', ''pytorch'',
        ''tensorflow.js'']}, {''software'': [''ilastik'', ''imagej'', ''fiji'', ''imjoy'',
        ''deepimagej'', ''n2v'']}, {''network'': [''unet'', ''densenet'', ''resnet'',
        ''inception'', ''shufflenet'']}, {''task'': [''semantic-segmentation'', ''instance-segmentation'',
        ''object-detection'', ''image-classification'', ''denoising'', ''image-restoration'',
        ''image-reconstruction'', ''in-silico-labeling'']}]'
      weights: {onnx: missing 'opset_version'}
  - {bioimageio_core_version: 0.5.3post2, bioimageio_spec_version: 0.4.5post4, error: null,
    name: reproduced test outputs from test inputs, status: passed, traceback: null}
  deepimagej:
  - bioimageio_core_version: 0.5.3post2
    bioimageio_spec_version: 0.4.5post7
    error: 'unsupported operand type(s) for +: ''PosixPath'' and ''str'''
    name: reproduce test outputs with deepimagej 2.1.13 using onnx weights
    reason: error
    status: failed
    traceback: ["  File \"/home/runner/work/models/models/scripts/check_rdf_and_test_model.py\",
        line 122, in test_resource\n    error = (error or \"\") + download_deepimagej_model(fiji_path,
        rdf_path)\n", "  File \"/home/runner/work/models/models/scripts/utils.py\",
        line 27, in download_deepimagej_model\n    os.makedirs(fiji_dir + \"//models\")\n"]
  ilastik:
  - bioimageio_core_version: 0.5.3post2
    bioimageio_spec_version: 0.4.5post6
    error: 'Failed to download https://zenodo.org/api/files/91895c21-4e8e-4cd0-98c3-1fae496b6068/densenet_model.onnx
      (504 Server Error: Gateway Time-out for url: https://zenodo.org/api/files/91895c21-4e8e-4cd0-98c3-1fae496b6068/densenet_model.onnx)'
    name: reproduce test outputs with ilastik <todo version> (draft) using onnx weights
    reason: error
    status: failed
    traceback: ["  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/core/resource_tests.py\",
        line 39, in test_model\n    model_rdf, weights_priority_order=None if weight_format
        is None else [weight_format]\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/core/resource_io/io_.py\",
        line 57, in load_resource_description\n    rd: ResourceDescription = resolve_raw_resource_description(raw_rd=raw_rd,
        nodes_module=nodes)\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/core/resource_io/utils.py\",
        line 112, in resolve_raw_resource_description\n    rd = UriNodeTransformer(root_path=raw_rd.root_path,
        uri_only_if_in_package=uri_only_if_in_package).transform(raw_rd)\n", "  File
        \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 54, in transform\n    return transformer(node, **kwargs)  # noqa\n",
      "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 86, in generic_transformer\n    node, **{n: self.transform(value, name=n,
        parent=node) for n, value in iter_fields(node)}\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 86, in <dictcomp>\n    node, **{n: self.transform(value, name=n, parent=node)
        for n, value in iter_fields(node)}\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 54, in transform\n    return transformer(node, **kwargs)  # noqa\n",
      "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 63, in transform_dict\n    return {key: self.transform(value, **kwargs)
        for key, value in node.items()}\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 63, in <dictcomp>\n    return {key: self.transform(value, **kwargs) for
        key, value in node.items()}\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 54, in transform\n    return transformer(node, **kwargs)  # noqa\n",
      "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 86, in generic_transformer\n    node, **{n: self.transform(value, name=n,
        parent=node) for n, value in iter_fields(node)}\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 86, in <dictcomp>\n    node, **{n: self.transform(value, name=n, parent=node)
        for n, value in iter_fields(node)}\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 54, in transform\n    return transformer(node, **kwargs)  # noqa\n",
      "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/node_transformer.py\",
        line 206, in transform_URI\n    local_path = _resolve_source(node, root_path=self.root_path)\n",
      "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/functools.py\",
        line 840, in wrapper\n    return dispatch(args[0].__class__)(*args, **kw)\n",
      "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/_resolve_source.py\",
        line 212, in _resolve_source_uri_node\n    local_path = _download_url(path_or_remote_uri,
        output)\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/spec/shared/_resolve_source.py\",
        line 396, in _download_url\n    raise RuntimeError(f\"Failed to download {uri}
        ({e})\")\n"]
  - bioimageio_core_version: 0.5.1
    bioimageio_spec_version: 0.4.5post3
    error: No module named 'onnxruntime'
    name: reproduce test outputs with ilastik <todo version> (draft) using onnx weights
    reason: error
    status: failed
    traceback: ["  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/core/resource_tests.py\",
        line 148, in test_resource\n    bioimageio_model=model, devices=devices, weight_format=weight_format\n",
      "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/core/prediction_pipeline/_prediction_pipeline.py\",
        line 159, in create_prediction_pipeline\n    bioimageio_model=bioimageio_model,
        devices=devices, weight_format=weight_format\n", "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/core/prediction_pipeline/_model_adapters/_model_adapter.py\",
        line 113, in create_model_adapter\n    adapter_cls = _get_model_adapter(weight)\n",
      "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/core/prediction_pipeline/_model_adapters/_model_adapter.py\",
        line 138, in _get_model_adapter\n    from ._onnx_model_adapter import ONNXModelAdapter\n",
      "  File \"/usr/share/miniconda3/envs/ilastik-env/lib/python3.7/site-packages/bioimageio/core/prediction_pipeline/_model_adapters/_onnx_model_adapter.py\",
        line 5, in <module>\n    import onnxruntime as rt\n"]
